lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,363 @@
+                # delete image
+                self.delete_image()
+                # the delete does not actually return anything. if no exception, then we'll assume
+                # it worked.
+                self.results['state']['status'] = 'Deleted'
+
+        return self.results
+
+    def image_to_dict(self, image):
+        result = self.serialize_obj(image, 'Image')
+        result['provisioning_state'] = image.provisioning_state
+        result['name'] = image.name
+        result['id'] = image.id
+        result['tags'] = image.tags or {}
+        return result
+
+    def resolve_storage_source(self, source):
+        blob_uri = None
+        disk = None
+        snapshot = None
+        if source.lower().endswith('.vhd'):
+            blob_uri = source
+            return (blob_uri, disk, snapshot)
+
+        tokenize = parse_resource_id(source)
+        if tokenize.get('type') == 'disks':
+            disk = source
+            return (blob_uri, disk, snapshot)
+
+        if tokenize.get('type') == 'snapshots':
+            snapshot = source
+            return (blob_uri, disk, snapshot)
+
+        # not a disk or snapshots
+        if 'type' in tokenize:
+            return (blob_uri, disk, snapshot)
+
+        # source can be name of snapshot or disk
+        snapshot_instance = self.get_snapshot(source)
+        if snapshot_instance:
+            snapshot = snapshot_instance.id
+            return (blob_uri, disk, snapshot)
+
+        disk_instance = self.get_disk(source)
+        if disk_instance:
+            disk = disk_instance.id
+        return (blob_uri, disk, snapshot)
+
+    def create_os_disk(self):
+        blob_uri, disk, snapshot = self.resolve_storage_source(self.source)
+        snapshot_resource = SubResource(snapshot) if snapshot else None
+        managed_disk = SubResource(disk) if disk else None
+        return ImageOSDisk(os_type=self.os_type,
+                           os_state=OperatingSystemStateTypes.generalized,
+                           snapshot=snapshot_resource,
+                           managed_disk=managed_disk,
+                           blob_uri=blob_uri)
+
+    def create_data_disks(self):
+        data_disks = []
+        lun = 0
+        if self.data_disk_sources:
+            for disk_source in self.data_disk_sources:
+                blob_uri, disk, snapshot = self.resolve_storage_source(disk_source)
+                if blob_uri or disk or snapshot:
+                    snapshot_resource = SubResource(snapshot) if snapshot else None
+                    managed_disk = SubResource(disk) if disk else None
+                    data_disks.append(ImageDataDisk(lun,
+                                    blob_uri=blob_uri,
+                                    snapshot=snapshot_resource,
+                                    managed_disk=managed_disk))
+                    lun += 1
+        return data_disks
+
+    def get_snapshot(self, snapshot_name):

```

Comment:
Minor suggestion.
The following 4 methods seem share the same pattern. You can have a more generic method which accepts getter method as its arguments.
Anyway up to u.